services/appveyor/appveyor-build.service.js

```diff
@@ -1,23 +1,31 @@
 import { renderBuildStatusBadge } from '../build-status.js'
+import { pathParam } from '../index.js'
 import AppVeyorBase from './appveyor-base.js'
 
 export default class AppVeyorBuild extends AppVeyorBase {
   static route = this.buildRoute('appveyor/build')
 
-  static examples = [
-    {
-      title: 'AppVeyor',
-      pattern: ':user/:repo',
-      namedParams: { user: 'gruntjs', repo: 'grunt' },
-      staticPreview: this.render({ status: 'success' }),
+  static openApi = {
+    '/appveyor/build/{user}/{repo}': {
+      get: {
+        summary: 'AppVeyor',
+        parameters: [
+          pathParam({ name: 'user', example: 'gruntjs' }),
+          pathParam({ name: 'repo', example: 'grunt' }),
+        ],
+      },
     },
-    {
-      title: 'AppVeyor branch',
-      pattern: ':user/:repo/:branch',
-      namedParams: { user: 'gruntjs', repo: 'grunt', branch: 'master' },
-      staticPreview: this.render({ status: 'success' }),
+    '/appveyor/build/{user}/{repo}/{branch}': {
+      get: {
+        summary: 'AppVeyor (with branch)',
+        parameters: [
+          pathParam({ name: 'user', example: 'gruntjs' }),
+          pathParam({ name: 'repo', example: 'grunt' }),
+          pathParam({ name: 'branch', example: 'master' }),
+        ],
```

Comment:
This has me wondering if it'd be worth either adding another wrapper function around `pathParam` that takes an array, e.g. `pathParams` that'll iterate over the entry args and contain the call sites to `pathParam`.

That in turn would allow this to be transformed into something like:

```suggestion
       parameters: pathParams([
          { name: 'user', example: 'gruntjs' },
          { name: 'repo', example: 'grunt' },
          { name: 'branch', example: 'master' },
        ]),
````

My thinking is that might help minimize some of the rightward drift (fewer characters/columns after all), and the potential for prettier to want to do some multi-line wrapping to account for the width.

Thoughts?

Reply:
Yes. Love it!

I'll add these for both `pathParams()` and `queryParams()`.

Having got into this...

One implementation of this would be that the plural variants take an array as you've suggested.
So the implementation would look like:

```js
function pathParams(params) {
  return params.map(param => pathParam(param))
}
```

..and then calling that would look like:

```js
pathParams([{ name: 'onlyParam', example: 'onlyValue' }])

pathParams([
  { name: 'param1', example: 'value1' },
  { name: 'param2', example: 'value2' },
])
```

The other option would be to have the plural variants take a spread as input instead of an array.
So then the implementation would look like:

```js
function pathParams(...params) {
  return params.map(param => pathParam(param))
}
```

..and calling that would look like:

```js
pathParams({ name: 'onlyParam', example: 'onlyValue' })

pathParams(
  { name: 'param1', example: 'value1' },
  { name: 'param2', example: 'value2' },
)
```

Having thought about it a bit, I prefer the second variant myself. What do you reckon?

Note to self: When you finish this off, these should also have DocStrings!

Reply:
Sure, second approach is my preference too :+1: